session-replay/Sources/DatadogSessionReplay/Processor/SRDataModelsBuilder/RecordsBuilder.swift

```diff
@@ -36,14 +36,62 @@ internal class RecordsBuilder {
         return .focusRecord(value: record)
     }
 
-    /// Creates Full Snapshot Record - bringing self-contained description of a single frame of the replay.
-    func createFullSnapshotRecord(from snapshot: ViewTreeSnapshot, with wireframes: [SRWireframe]) -> SRRecord {
+    // MARK: - Creating FSR and ISR
+
+    /// Wireframes from last FSR or ISR.
+    private var lastWireframes: [SRWireframe]?
+
+    func createFullOrIncrementalSnapshotRecord(from snapshot: ViewTreeSnapshot, with wireframes: [SRWireframe]) -> SRRecord? {
+        defer { lastWireframes = wireframes }
+
+        if let lastWireframes = lastWireframes {
+            do {
+                return try createIncrementalRecord(from: snapshot, newWireframes: wireframes, lastWireframes: lastWireframes)
+            } catch { // TODO: RUMM-2410 Use `DD.logger` and / or `DD.telemetry` to report ISR errors
+                // In case of any trouble, fallback to FSR which is always possible:
+                return createFullSnapshotRecord(from: snapshot, wireframes: wireframes)
+            }
+        } else {
+            return createFullSnapshotRecord(from: snapshot, wireframes: wireframes)
+        }
```

Comment:
`guard` would be best here :)

Reply:
Done 👍👌!

Although, I prefer using `guard x() else { y() }` to express that _"we **expect** `x`, but if it's not there, **fallback** to `y()`"_. In this case `createFSR()` is not a fallback but a regular control flow (_"if `lastWireframes` is not yet there, produce FSR"_). However, it is just my habit and not a shared convention, so I changed it to `guard` which we use more often for similar things.